packages/server/graphql/mutations/helpers/generateDiscussionSummary.ts

```diff
@@ -5,19 +5,21 @@ import updateDiscussions from '../../../postgres/queries/updateDiscussions'
 import OpenAIServerManager from '../../../utils/OpenAIServerManager'
 import publish from '../../../utils/publish'
 import {DataLoaderWorker} from '../../graphql'
+import canAccessAISummary from './canAccessAISummary'
 
 const generateDiscussionSummary = async (
   discussionId: string,
   meeting: MeetingRetrospective,
   dataLoader: DataLoaderWorker
 ) => {
-  const {id: meetingId, endedAt, facilitatorUserId} = meeting
-  const [facilitator, comments, tasks] = await Promise.all([
+  const {id: meetingId, endedAt, facilitatorUserId, teamId} = meeting
+  const [facilitator, comments, tasks, team] = await Promise.all([
     dataLoader.get('users').loadNonNull(facilitatorUserId),
     dataLoader.get('commentsByDiscussionId').load(discussionId),
-    dataLoader.get('tasksByDiscussionId').load(discussionId)
+    dataLoader.get('tasksByDiscussionId').load(discussionId),
+    dataLoader.get('teams').load(teamId)
   ])
-  if (!facilitator.featureFlags.includes('aiSummary')) return
+  if (!canAccessAISummary(team, facilitator.featureFlags)) return
```

Comment:
As far as I understand, in most of cases a team can not access AI summary, should we add some checks before loading bunch of data in `Promise.all`?

Reply:
Yeah, I was thinking about this, but to check `canAccessAISummary` we need to grab the `facilitator` and the `team`. We could await those two fields, and if they pass the `canAccessAISummary` check, we grab the `comments` and `tasks` in the discussion thread, but as we're only getting them from one `discussionId`, they should be relatively cheap queries. 

It's possible that Rethink is super slow with one of the requests, but then separating the `comments` and `tasks` requests will normally mean a slightly slower response time for AI Summary users, which will hopefully be a majority of users soon. I don't have a strong preference but lean towards the current implementation. What do you think?

Reply:
Make sense, then keeping it as is is fine I think

Reply:
+1 I'm not so sure about this as this repeats in the other files. It looks like we wouldn't load most of the data if they don't qualify for AI summaries, also not in the context where this function is called. In this case I would err on the side of caution and load it only for qualified users.